planner/cascades/implementation_rules.go

```diff
@@ -64,6 +64,10 @@ var defaultImplementationMap = map[memo.Operand][]ImplementationRule{
 		&ImplTopN{},
 		&ImplTopNAsLimit{},
 	},
+	memo.OperandJoin: {
+		&ImplHashJoinBuildLeft{},
+		&ImplHashJoinBuildRight{},
```

Comment:
Are the `outer` and `inner` better than the names `left` and `right`?

Reply:
For other join types, `outer` and `inner` is better. But I think it doesn't make sense for `InnerJoin`. While, `left` and `right` have meaning for all of them.